src/sentry/db/router.py

```diff
@@ -0,0 +1,126 @@
+import sys
+from typing import List
+
+from django.apps import apps
+from django.conf import settings
+from django.db import connections
+from django.db.utils import ConnectionDoesNotExist
+
+from sentry.db.models.base import Model
+from sentry.silo.base import SiloMode
+
+
+class SiloRouter:
+    """
+    Django database router for multi-region deployments.
+
+    We support two configurations:
+
+    - Monolith - all tables reside in the same database.
+    - Siloed - tables for control and region are separated.
+
+    Within Siloed there are are two flavours:
+
+    - simulated - If the application is configured with `control` and `region`
+      connections, then we are in 'simulated' silo environment (like our testsuite).
+    - isolated - If there are no control/region connections we map the `default`
+      connection to be the region/control database and assume the
+      'other' silo is inaccessible.
+    """
+
+    __simulated_map = {
+        SiloMode.MONOLITH: "default",
+        SiloMode.REGION: "region",
+        SiloMode.CONTROL: "control",
+    }
+
+    __table_to_silo = {}
+
+    __is_simulated = False
+    """Whether or not we're operating in a simulated silo environment"""
+
+    def __init__(self):
+        try:
+            # By accessing the connections Django will raise
+            # Use `assert` to appease linters
+            assert connections["control"]
+            assert connections["region"]
+            self.__is_simulated = True
+        except (AssertionError, ConnectionDoesNotExist):
+            self.__is_simulated = False
+
+    def use_simulated(self, value: bool):
+        if "pytest" not in sys.modules:
+            raise ValueError("Cannot mutate simulation mode outside of tests")
+        self.__is_simulated = value
+
+    def _resolve_silo_connection(self, silo_modes: List[SiloMode], table: str):
+        active_mode = settings.SILO_MODE
```

Comment:
Is this safe?  Should we use `SiloMode.get_current()`?  Concern is for our integration tests which use thread level manipulation of the current SiloMode.

Reply:
Good catch. I'll get that updated.